tests/unit/providers/test_xai_provider.py

```
@@ -169,3 +170,104 @@ def test_per_request_timeout_is_declared_unsupported() -> None:
+
+
+def test_clients_for_closed_loops_are_dropped_and_closed() -> None:
+    """Sync callers get a throwaway loop per request, so the per-loop cache has to shed the
+    dead ones. Left alone it would hold a live grpc channel, and its socket, per call made
+    for the whole life of the provider.
+    """
+    from any_llm.providers.xai.xai import XaiProvider
+
+    built: list[MagicMock] = []
+
+    def _build(**_: Any) -> MagicMock:
```

Ruff (0.16.1)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `**_`

(ANN401)